### PR TITLE
Cleaner CinemaOT

### PR DESCRIPTION
**Pull request by eroell**
**Description of changes**
Vectorizing for loops, remove redundant code.

**Technical details**
Don't have large memory machine available atm for larger benchmark check.
For small 30k samples, `mean_absolute`, `x_ordered`, `correlation` become ~0.2 to 0.4 seconds instead of ~2 to 3 seconds, in overall 20seconds runtime. Sinkhorn dominating runtime.

**Additional context**

### CODE REVIEW

**Review by Zethson — COMMENTED** [review]
Lots of good simplifications! I assume that the results are the same. Can you ott Jax jit the computation though? That should be the primary loss of performance.

**Comment by eroell** [comment]
> I assume that the results are the same.

While we don't have tight CI tests here, verified with the snippet below that the two implementations generate the same result, by comparing the following four saved outputs from before and with this change, which pass a `np.allclose` check.

i.e. running this script once on main, once on current branch
```py
import numpy as np
import pertpy as pt
import scanpy as sc

adata = pt.dt.cinemaot_example()
sc.pp.sample(adata, n=1_000, replace=True, rng=42)
adata.X = adata.raw.X.copy()

sc.pp.pca(adata)

cot = pt.tl.Cinemaot()

de = cot.causaleffect(
    adata,
    pert_key="perturbation",
    control="No stimulation",
    return_matching=True,
    thres=0.34, # a threshold where for this dataset, the implementation of Xi matters
    smoothness=3e-5,
    eps=1e-3,
    solver="Sinkhorn",
    preweight_label="cell_type0528",
)

np.save("cinemaot_<branch_name>_cft.npy", adata.obsm["cf"])
np.save("cinemaot_<branch_name>_X.npy", de.X)
np.save("cinemaot_<branch_name>_embedding.npy", de.obsm["X_embedding"])
np.save("cinemaot_<branch_name>_ot.npy", de.obsm["ot"])
```

**Comment by eroell** [comment]
> Can you ott Jax jit the computation though? That should be the primary loss of performance.

It is. My attempts here on speeding it up have not been successful. The recommendation e.g. in their tutorial [here](https://ott-jax.readthedocs.io/en/latest/tutorials/basic_ot_between_datasets.html) does not give speedups when their LinearProblem has parameters `a` and `b`. It seems to give a slight speedup without `a`, `b` though, but this is just ca ~10% faster (across a range of sample sizes).
